partner_invoicing_mode_at_shipping/models/res_partner.py

```diff
@@ -11,3 +12,33 @@ class ResPartner(models.Model):
         selection_add=[("at_shipping", "At Shipping")],
         ondelete={"at_shipping": "set default"},
     )
+
+    one_invoice_per_picking = fields.Boolean(
+        help="Check this if you want to create one invoice per picking using the"
+        " partner invoicing mode that should be different than 'At Shipping'."
+    )
+
+    @api.constrains(
+        "invoicing_mode", "one_invoice_per_picking", "one_invoice_per_order"
+    )
```

Comment:
Why not making `one_invoice_per_shipping` computed store=True and set it False when invoicing_mode is "At shipping" or one_invoice_per_order is set ?

Reply:
that could be an improvement yes